### ISSUE TITLE
`helmfile apply` concurrent map iteration and map write

### ISSUE THREAD

**Issue by andrascz** [roboll/helmfile#1495]
After upgrading to 0.130.0 I got some panics about concurrent map iteration/write when running `helmfile apply`, `helmfile destroy` works.
The same issue is there when downgrading to 0.129.4
0.129.3 works fine.

Stacktrace:
```
fatal error: concurrent map iteration and map write

goroutine 924 [running]:
runtime.throw(0x1651e5b, 0x26)
        /usr/local/go/src/runtime/panic.go:1116 +0x72 fp=0xc000746728 sp=0xc0007466f8 pc=0x433182
runtime.mapiternext(0xc000516720)
        /usr/local/go/src/runtime/map.go:853 +0x552 fp=0xc0007467a8 sp=0xc000746728 pc=0x40f542
runtime.mapiterinit(0x14227e0, 0xc000197590, 0xc000516720)
        /usr/local/go/src/runtime/map.go:843 +0x1c4 fp=0xc0007467c8 sp=0xc0007467a8 pc=0x40eef4
reflect.mapiterinit(0x14227e0, 0xc000197590, 0x14227e0)
        /usr/local/go/src/runtime/map.go:1331 +0x54 fp=0xc0007467f8 sp=0xc0007467c8 pc=0x4102f4
reflect.Value.MapKeys(0x14227e0, 0xc000197590, 0x15, 0x1374aa0, 0xc00061fbf0, 0x98)
        /usr/local/go/src/reflect/value.go:1192 +0xc5 fp=0xc000746888 sp=0xc0007467f8 pc=0x49c545
github.com/imdario/mergo.deepMerge(0x14227e0, 0xc000197590, 0x15, 0x14227e0, 0xc000197590, 0x15, 0xc000747a28, 0x5, 0xc00046fc40, 0x0, ...)
        /home/circleci/workspace/helmfile/vendor/github.com/imdario/mergo/merge.go:114 +0x320 fp=0xc000746b58 sp=0xc000746888 pc=0xc71920
github.com/imdario/mergo.deepMerge(0x14227e0, 0xc000197500, 0x15, 0x14227e0, 0xc000197500, 0x15, 0xc000747a28, 0x4, 0xc00046fc40, 0x0, ...)
        /home/circleci/workspace/helmfile/vendor/github.com/imdario/mergo/merge.go:147 +0xa67 fp=0xc000746e28 sp=0xc000746b58 pc=0xc72067
github.com/imdario/mergo.deepMerge(0x14227e0, 0xc0001974d0, 0x15, 0x14227e0, 0xc0001974d0, 0x15, 0xc000747a28, 0x3, 0xc00046fc40, 0x0, ...)
        /home/circleci/workspace/helmfile/vendor/github.com/imdario/mergo/merge.go:147 +0xa67 fp=0xc0007470f8 sp=0xc000746e28 pc=0xc72067
github.com/imdario/mergo.deepMerge(0x14227e0, 0xc000196e70, 0x15, 0x14227e0, 0xc000196e70, 0x15, 0xc000747a28, 0x2, 0xc00046fc40, 0x0, ...)
        /home/circleci/workspace/helmfile/vendor/github.com/imdario/mergo/merge.go:147 +0xa67 fp=0xc0007473c8 sp=0xc0007470f8 pc=0xc72067
github.com/imdario/mergo.deepMerge(0x14227e0, 0xc000197b00, 0x15, 0x14227e0, 0xc000196bd0, 0x15, 0xc000747a28, 0x1, 0xc00046fc40, 0x0, ...)
        /home/circleci/workspace/helmfile/vendor/github.com/imdario/mergo/merge.go:147 +0xa67 fp=0xc000747698 sp=0xc0007473c8 pc=0xc72067
github.com/imdario/mergo.deepMerge(0x14227e0, 0xc00063e078, 0x195, 0x14227e0, 0xc0006eb830, 0x15, 0xc000747a28, 0x0, 0xc00046fc40, 0x15, ...)
        /home/circleci/workspace/helmfile/vendor/github.com/imdario/mergo/merge.go:147 +0xa67 fp=0xc000747968 sp=0xc000747698 pc=0xc72067
github.com/imdario/mergo.merge(0x13448e0, 0xc00063e078, 0x14227e0, 0xc0006eb830, 0xc000747b48, 0x2, 0x2, 0x0, 0x0)
        /home/circleci/workspace/helmfile/vendor/github.com/imdario/mergo/merge.go:361 +0x32a fp=0xc000747af8 sp=0xc000747968 pc=0xc74eda
github.com/imdario/mergo.Merge(...)
        /home/circleci/workspace/helmfile/vendor/github.com/imdario/mergo/merge.go:291
github.com/roboll/helmfile/pkg/environment.(*Environment).GetMergedValues(0xc000422bf0, 0x0, 0x0, 0x0)
        /home/circleci/workspace/helmfile/pkg/environment/environment.go:75 +0x14d fp=0xc000747b78 sp=0xc000747af8 pc=0xc7800d
github.com/roboll/helmfile/pkg/state.(*HelmState).Values(...)
        /home/circleci/workspace/helmfile/pkg/state/state_exec_tmpl.go:12
github.com/roboll/helmfile/pkg/state.(*HelmState).triggerReleaseEvent(0xc000422a00, 0x162613b, 0x7, 0x0, 0x0, 0xc00052c000, 0x1622ece, 0x5, 0x435c9c, 0xc0002a45f8, ...)
        /home/circleci/workspace/helmfile/pkg/state/state.go:1888 +0x178 fp=0xc000747da8 sp=0xc000747b78 pc=0x1237e08
github.com/roboll/helmfile/pkg/state.(*HelmState).triggerPrepareEvent(...)
        /home/circleci/workspace/helmfile/pkg/state/state.go:1863
github.com/roboll/helmfile/pkg/state.(*HelmState).PrepareCharts.func2(0x2)
        /home/circleci/workspace/helmfile/pkg/state/state.go:886 +0x18b fp=0xc000747fa8 sp=0xc000747da8 pc=0x124b81b
github.com/roboll/helmfile/pkg/state.(*HelmState).scatterGather.func1(0xc00007b7a0, 0xc00061b700, 0x2)
        /home/circleci/workspace/helmfile/pkg/state/state_run.go:41 +0x30 fp=0xc000747fc8 sp=0xc000747fa8 pc=0x1250cc0
runtime.goexit()
        /usr/local/go/src/runtime/asm_amd64.s:1373 +0x1 fp=0xc000747fd0 sp=0xc000747fc8 pc=0x463211
created by github.com/roboll/helmfile/pkg/state.(*HelmState).scatterGather
        /home/circleci/workspace/helmfile/pkg/state/state_run.go:40 +0x171

goroutine 1 [chan receive]:
github.com/roboll/helmfile/pkg/state.(*HelmState).PrepareCharts.func3()
        /home/circleci/workspace/helmfile/pkg/state/state.go:1017 +0xbb
github.com/roboll/helmfile/pkg/state.(*HelmState).scatterGather(0xc000422a00, 0x2, 0x10, 0xc000305110, 0xc00007b7a0, 0xc0008579b8)
        /home/circleci/workspace/helmfile/pkg/state/state_run.go:46 +0x193
github.com/roboll/helmfile/pkg/state.(*HelmState).PrepareCharts(0xc000422a00, 0x1bcb140, 0xc000126000, 0xc00061b6a0, 0xe, 0x2, 0x1622ece, 0x5, 0x0, 0x0, ...)
        /home/circleci/workspace/helmfile/pkg/state/state.go:870 +0x364
github.com/roboll/helmfile/pkg/app.(*Run).withPreparedCharts(0xc000304570, 0x1622ece, 0x5, 0x1000000, 0xc000857b70, 0x0, 0x0)
        /home/circleci/workspace/helmfile/pkg/app/run.go:62 +0x1b3
github.com/roboll/helmfile/pkg/app.(*App).Apply.func1(0xc000304570, 0xc000304500, 0x0, 0x0, 0x0)
        /home/circleci/workspace/helmfile/pkg/app/app.go:302 +0x169
github.com/roboll/helmfile/pkg/app.(*App).ForEachState.func1(0xc000422a00, 0xc000422a00, 0x0, 0x0, 0xc000422780)
        /home/circleci/workspace/helmfile/pkg/app/app.go:788 +0xc4
github.com/roboll/helmfile/pkg/app.(*App).visitStates.func1(0xc000039b8b, 0xf, 0xc0001d4700, 0x38, 0x0, 0x0)
        /home/circleci/workspace/helmfile/pkg/app/app.go:736 +0x7ec
github.com/roboll/helmfile/pkg/app.(*App).visitStateFiles.func1(0xc0004648d0, 0x2d)
        /home/circleci/workspace/helmfile/pkg/app/app.go:570 +0x9d
github.com/roboll/helmfile/pkg/app.(*App).within(0xc00011de60, 0xc000039b80, 0xa, 0xc000858290, 0xc000858250, 0x2)
        /home/circleci/workspace/helmfile/pkg/app/app.go:531 +0x33e
github.com/roboll/helmfile/pkg/app.(*App).visitStateFiles(0xc00011de60, 0x0, 0x0, 0x2628170, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1, ...)
        /home/circleci/workspace/helmfile/pkg/app/app.go:564 +0x2f2
github.com/roboll/helmfile/pkg/app.(*App).visitStates(0xc00011de60, 0x0, 0x0, 0x2628170, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1, ...)
        /home/circleci/workspace/helmfile/pkg/app/app.go:648 +0x10b
github.com/roboll/helmfile/pkg/app.(*App).visitStatesWithSelectorsAndRemoteSupport(0xc00011de60, 0x0, 0x0, 0xc00079e5b0, 0xc0004b6bc0, 0x1, 0x1, 0x1, 0x0)
        /home/circleci/workspace/helmfile/pkg/app/app.go:903 +0x4c1
github.com/roboll/helmfile/pkg/app.(*App).ForEachState(0xc00011de60, 0xc00079e640, 0xc0004b6bc0, 0x1, 0x1, 0xc0004b6bc0, 0x0)
        /home/circleci/workspace/helmfile/pkg/app/app.go:784 +0xb2
github.com/roboll/helmfile/pkg/app.(*App).Apply(0xc00011de60, 0x1bc6320, 0xc000079a60, 0xc000079a60, 0xc0004b6bb8)
        /home/circleci/workspace/helmfile/pkg/app/app.go:301 +0x15b
main.main.func9(0xc00011de60, 0xc00014f080, 0x0, 0xc000079a50, 0x0)
        /home/circleci/workspace/helmfile/main.go:405 +0x6d
main.action.func1(0xc00014f080, 0x0, 0xc00014f080)
        /home/circleci/workspace/helmfile/main.go:744 +0xd4
github.com/urfave/cli.HandleAction(0x13979e0, 0xc000078820, 0xc00014f080, 0xc0000aa700, 0x0)
        /home/circleci/workspace/helmfile/vendor/github.com/urfave/cli/app.go:490 +0xc8
github.com/urfave/cli.Command.Run(0x1622ece, 0x5, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1666b42, 0x3f, 0x0, ...)
        /home/circleci/workspace/helmfile/vendor/github.com/urfave/cli/command.go:210 +0x9e8
github.com/urfave/cli.(*App).Run(0xc000102340, 0xc00000c080, 0x2, 0x2, 0x0, 0x0)
        /home/circleci/workspace/helmfile/vendor/github.com/urfave/cli/app.go:255 +0x741
main.main()
        /home/circleci/workspace/helmfile/main.go:535 +0x39b4

goroutine 6 [select]:
go.opencensus.io/stats/view.(*worker).start(0xc000126780)
        /home/circleci/workspace/helmfile/vendor/go.opencensus.io/stats/view/worker.go:276 +0x100
created by go.opencensus.io/stats/view.init.0
        /home/circleci/workspace/helmfile/vendor/go.opencensus.io/stats/view/worker.go:34 +0x68

goroutine 65 [syscall, 1 minutes]:
os/signal.signal_recv(0x0)
        /usr/local/go/src/runtime/sigqueue.go:147 +0x9c
os/signal.loop()
        /usr/local/go/src/os/signal/signal_unix.go:23 +0x22
created by os/signal.Notify.func1
        /usr/local/go/src/os/signal/signal.go:127 +0x44

goroutine 36 [chan receive, 1 minutes]:
github.com/roboll/helmfile/pkg/app.(*App).visitStates.func1.1(0xc000714900, 0xc00011de60, 0xc000186500, 0x2628170, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1, ...)
        /home/circleci/workspace/helmfile/pkg/app/app.go:660 +0x52
created by github.com/roboll/helmfile/pkg/app.(*App).visitStates.func1
        /home/circleci/workspace/helmfile/pkg/app/app.go:659 +0x2b5

goroutine 376 [chan receive]:
github.com/roboll/helmfile/pkg/app.(*App).visitStates.func1.1(0xc00077b560, 0xc00011de60, 0xc000422780, 0x2628170, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1, ...)
        /home/circleci/workspace/helmfile/pkg/app/app.go:660 +0x52
created by github.com/roboll/helmfile/pkg/app.(*App).visitStates.func1
        /home/circleci/workspace/helmfile/pkg/app/app.go:659 +0x2b5

goroutine 923 [runnable]:
reflect.typedmemmove(0x13f9ea0, 0xc0003007f0, 0xc0007e74c8)
        /usr/local/go/src/runtime/mbarrier.go:177 +0x4e
reflect.copyVal(0x13f9ea0, 0x14, 0xc0007e74c8, 0xc0007e74c8, 0x16, 0x1374aa0)
        /usr/local/go/src/reflect/value.go:1287 +0x5a
reflect.Value.MapIndex(0x14227e0, 0xc0001976e0, 0x15, 0x1374aa0, 0xc0003007b0, 0x98, 0x13f9ea0, 0xc0003007d0, 0x94)
        /usr/local/go/src/reflect/value.go:1173 +0x1c5
github.com/imdario/mergo.deepMerge(0x14227e0, 0xc0001976e0, 0x15, 0x14227e0, 0xc0001976e0, 0x15, 0xc0000dfa28, 0x2, 0xc00044a040, 0x0, ...)
        /home/circleci/workspace/helmfile/vendor/github.com/imdario/mergo/merge.go:119 +0x4a2
github.com/imdario/mergo.deepMerge(0x14227e0, 0xc000197b00, 0x15, 0x14227e0, 0xc000196bd0, 0x15, 0xc0000dfa28, 0x1, 0xc00044a040, 0x0, ...)
        /home/circleci/workspace/helmfile/vendor/github.com/imdario/mergo/merge.go:147 +0xa67
github.com/imdario/mergo.deepMerge(0x14227e0, 0xc00000e030, 0x195, 0x14227e0, 0xc0006eb830, 0x15, 0xc0000dfa28, 0x0, 0xc00044a040, 0x15, ...)
        /home/circleci/workspace/helmfile/vendor/github.com/imdario/mergo/merge.go:147 +0xa67
github.com/imdario/mergo.merge(0x13448e0, 0xc00000e030, 0x14227e0, 0xc0006eb830, 0xc0000dfb48, 0x2, 0x2, 0x0, 0x0)
        /home/circleci/workspace/helmfile/vendor/github.com/imdario/mergo/merge.go:361 +0x32a
github.com/imdario/mergo.Merge(...)
        /home/circleci/workspace/helmfile/vendor/github.com/imdario/mergo/merge.go:291
github.com/roboll/helmfile/pkg/environment.(*Environment).GetMergedValues(0xc000422bf0, 0x0, 0x0, 0x0)
        /home/circleci/workspace/helmfile/pkg/environment/environment.go:75 +0x14d
github.com/roboll/helmfile/pkg/state.(*HelmState).Values(...)
        /home/circleci/workspace/helmfile/pkg/state/state_exec_tmpl.go:12
github.com/roboll/helmfile/pkg/state.(*HelmState).triggerReleaseEvent(0xc000422a00, 0x162613b, 0x7, 0x0, 0x0, 0xc00052c2b0, 0x1622ece, 0x5, 0x435c9c, 0xc0002a2df8, ...)
        /home/circleci/workspace/helmfile/pkg/state/state.go:1888 +0x178
github.com/roboll/helmfile/pkg/state.(*HelmState).triggerPrepareEvent(...)
        /home/circleci/workspace/helmfile/pkg/state/state.go:1863
github.com/roboll/helmfile/pkg/state.(*HelmState).PrepareCharts.func2(0x1)
        /home/circleci/workspace/helmfile/pkg/state/state.go:886 +0x18b
github.com/roboll/helmfile/pkg/state.(*HelmState).scatterGather.func1(0xc00007b7a0, 0xc00061b700, 0x1)
        /home/circleci/workspace/helmfile/pkg/state/state_run.go:41 +0x30
created by github.com/roboll/helmfile/pkg/state.(*HelmState).scatterGather
        /home/circleci/workspace/helmfile/pkg/state/state_run.go:40 +0x171
```

**Comment by mumoshu** [comment]
@andrascz Thanks for reporting 🙏 This seems like a regression introduced in #1485. Let me fix this soon.

**Comment by mumoshu** [comment]
This is being fixed via #1534
